internal/command/versions.go

```diff
@@ -0,0 +1,48 @@
+package command
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+
+	"github.com/cyberark/summon/pkg/summon"
+	"github.com/cyberark/summon/provider"
+)
+//Create the AllVersions string to include versions of all the providers
+const SUMMON_VERSION = summon.VERSION
+var DefaultPath = provider.DefaultPath
+var AllVersions = GetVersions(GetProviderFileNames())
+
+//Run `$provider --version`(from DefaultPath),
+// 	for each provider in the list of providers
+//Then appends the responded versions to string and returns it
+func GetVersions(providers []string) string{
+	var toReturn = SUMMON_VERSION
+	for _, provider := range providers {
+		version, err := exec.Command(path.Join(DefaultPath, provider), "--version").Output()
+		if err != nil{
+			toReturn = toReturn + "\n"+provider+" version "+"ERROR"
+			fmt.Println("ERROR: "+provider, err, "running command :"+DefaultPath+"/"+ provider, "--version")	//debug
+			continue
+		}
+		toReturn = toReturn + "\n"+provider+" version "+string(version[0:len(version)-1])
```

Comment:
Avoid composing strings manually like this. If you are just trying to assemble some strings together in a repetitive manner, you should make a buffer and write to it like you would do it to a file.

Reply:
Pt2: What are you stripping from the end of the version string? In Linux it's one char for a newline but in windows it's two.

Reply:
I'm stripping a new line character from the end of the string

Reply:
I changed the way I made the string and stopped stripping the new line